### DISCUSSION TITLE
Implicit Cache When Performing Many Updates ?

### DISCUSSION THREAD

**Discussion by khazamov** [community/community#40471] [Actions] [answered]
### Select Topic Area

Bug

### Body

Hello github community, 

I noticed some differences between execution in GitHub Actions and locally running workflow under nektos/act when running my workflow script.  I attribute this difference to the  implicit cache. I do not use any caching functionality provided by GitHub. 

**How to reproduce:**

The last run of https://github.com/khazamov/testapi2/actions/runs/3585603687/jobs/6034223880 resulted in an unexpected result.  Notice in logs _GET /collections?workspace=None HTTP/1.1" 401 error._ 

"None" part in the URL comes from undefined variable. This was a buggy version of the script, which Github Actions seem to have cached. (see "Push Postman collections" step).

The repo is identical on my local machine, which has python 3.9 installed. Workspace variable  and consequently GET /collections?workspace=xxx request is properly defined when I run the script locally. 
In addition, I ran the workflow with nektos/act on my MacOS using Docker. Workspace variable and consequently GET /collections?workspace=xxx request is also properly defined in this case of running MacOS.

I wonder if anyone noticed a similar behavior?

I believe there is some strange caching issue (perhaps it's my fault for making  too many commits  and not testing everything locally). 

Is this something you've seen before?

**Comment by airtower-luna** [comment, marked as answer]
Looking at your script, the problem appears to be coming from [this line](https://github.com/khazamov/testapi2/blob/3bd1a4ff006ddc1307c657ef0b132181dc554a78/push_collection_to_repo.py#L15):

```py
workspaceId = os.getenv('POSTMAN_WORKSPACE_ID')
```

It will return `None` if the environment variable isn't defined. And I don't see anything in [your workflow](https://github.com/khazamov/testapi2/actions/runs/3585603687/workflow) that defines that variable.

I wonder if you have `POSTMAN_WORKSPACE_ID` set in your local setup? Mind that each step gets a fresh shell, environment variables set in one step won't carry over to others. If you want that, you need to [use `GITHUB_ENV`](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files). :thinking: 

Edit:
> I believe there is some strange caching issue (perhaps it's my fault for making too many commits and not testing everything locally).

I should emphasize that the script I linked is at the commit the workflow reports it ran for. If you didn't mean to run that version, the issue may be with *how* the workflow was triggered.

**Reply by flwajaber** [reply]
3

**Comment by khazamov** [comment]
thanks. Turns out i was using "re-run" workflow instead of running a new workflow. Your answer was helpful!
